### DISCUSSION TITLE
Execute a stored procedure after Serenity update

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6630] [Help]
Please is it possible to execute a stored procedure to the database after Serenity Update is completed possibly inside the repository class
or anywhere appropriate, I dont want to use sql triggers

`public SaveResponse Update(IUnitOfWork uow, SaveRequest<MyRow> request)`
        `{`
            `return new MySaveHandler(Context).Process(uow, request, SaveRequestType.Update);`   
        `}`

![image](https://user-images.githubusercontent.com/50826372/209577988-450b1c3a-d2e4-4b64-b691-ba8d865b66ad.png)

**Comment by stixoffire** [comment]
Why not use triggers ? 
And yes you can in Serenity . Look at the manual for your handlers. 
One caveat about serenity if the application gets closed before your other code executes; well too bad for your data base it will not be updated as you think it should because that code does not run. How many users and the possibilities that can occur for errors .. the possibilities multiply by the number of users and transactions.

Triggers work . I have used them many times - they run because your SQL code ran on the table and are not dependent on the application itself from multiple users to continue to run - for example a long transaction time from button click and I close application too bad for your database.
